### PR TITLE
** DO NOT MERGE THIS ** Fixing PropTypes & setting defaultProps

### PR DESCRIPTION
**Pull request by CodingWithCDJE**
* Modifying propTypes for certain files
* Set defaultProps to allow refresh

### CODE REVIEW

File: src/components/ProjectDetailView/ProjectDetailView.js [hunk 1]
```diff
@@ -47,28 +46,46 @@ ProjectDetailView.propTypes = {
...
+ProjectDetailView.defaultProps = {
+  match: {
+    params: {
+      projectname: '',
+    },
+    url: '',
+  },
+
+  projectDetail: {
+    id: 3,
```
Comment (horseeyephil) [inline src/components/ProjectDetailView/ProjectDetailView.js]: @CodingWithCDJE  good use of `defaultProps` - but putting `id: 3` isn't going to be a long term solution

File: src/components/ProjectBlockers/ProjectBlockers.js [hunk 2]
```diff
@@ -17,17 +17,15 @@ class ProjectBlockers extends Component {
...
-      .get('http://localhost:5000/api/blockers')
-      .then((result) => {
-        const newResult = result.data.filter((data) => {
-          if (projectID === data.projectID) {
-            return data;
-          }
-          return null;
-        });
-        this.setState({ blockers: newResult });
+    axios.get('http://localhost:5000/api/blockers').then((result) => {
```
Comment (horseeyephil) [inline src/components/ProjectBlockers/ProjectBlockers.js]: @cchanyu  - why are we requesting `All` blockers here and then filtering on the front-end??
Think about our API design - part of the contract is we should be able to request just the blockers associated with a given project!

File: src/components/ProjectBlockers/ProjectBlockers.js [hunk 3]
```diff
@@ -17,17 +17,15 @@ class ProjectBlockers extends Component {
...
+      const newResult = result.data.filter((data) => {
+        if (projectID === data.projectID) {
+          return data;
+        }
+        return null;
       });
+      this.setState({ blockers: newResult });
+    });
     axios
       .get(`http://localhost:5000/api/projects/${projectID}/users`)
```
Comment (horseeyephil) [inline src/components/ProjectBlockers/ProjectBlockers.js]: also - why are we requesting project members in the `Blockers` component? 😅

**Comment by CodingWithCDJE** [comment]
Do NOT MERGE THIS

**Pull request closed without merging**